# HG changeset patch
# Parent  1ccc537e56b330c85712a64764593c84d83d938a
Display a valid target as the label when a bulb has no label

When a bulb has no label lightsd displays its id (MAC address) instead.
This changeset keeps this behavior but removes the ":" from the MAC
address so it's a valid target. This yields to a slightly better user
experience when on-boarding new bulbs which might not have a label.

diff --git a/core/proto.c b/core/proto.c
--- a/core/proto.c
+++ b/core/proto.c
@@ -320,7 +320,7 @@
         (src), (start), (stop), (dst), sizeof((dst))    \
     )
 
-        char h[16], s[16], b[16];
+        char h[16], s[16], b[16], bulb_id[16];
         PRINT_COMPONENT(bulb->state.hue, h, 0, 360);
         PRINT_COMPONENT(bulb->state.saturation, s, 0, 1);
         PRINT_COMPONENT(bulb->state.brightness, b, 0, 1);
@@ -331,8 +331,14 @@
             label = bulb->state.label;
             label_size = (int)sizeof(bulb->state.label);
         } else {
-            label = bulb_addr;
-            label_size = (int)sizeof(bulb_addr);
+            label = bulb_id;
+            label_size = LGTD_ARRAY_SIZE(bulb_id);
+            snprintf(
+                bulb_id, label_size,
+                "%02hhx%02hhx%02hhx%02hhx%02hhx%02hhx",
+                bulb->addr[0], bulb->addr[1], bulb->addr[2],
+                bulb->addr[3], bulb->addr[4], bulb->addr[5]
+            );
         }
 
         LGTD_SNPRINTF_APPEND(
diff --git a/tests/core/proto/test_proto_get_light_state.c b/tests/core/proto/test_proto_get_light_state.c
--- a/tests/core/proto/test_proto_get_light_state.c
+++ b/tests/core/proto/test_proto_get_light_state.c
@@ -156,7 +156,7 @@
             "\"_vendor\":\"martine\","
             "\"hsbk\":[0,0,1,4000],"
             "\"power\":false,"
-            "\"label\":\"05:04:03:02:01:00\","
+            "\"label\":\"050403020100\","
             "\"tags\":[\"vapor\",\"d^-^b\"]"
         "},{"
             "\"_lifx\":{"
@@ -243,7 +243,7 @@
             "\"_vendor\":\"martine\","
             "\"hsbk\":[0,0,1,4000],"
             "\"power\":false,"
-            "\"label\":\"05:04:03:02:01:00\","
+            "\"label\":\"050403020100\","
             "\"tags\":[\"vapor\",\"d^-^b\"]"
         "},"
         "{"
diff --git a/tests/core/proto/test_proto_get_light_state_unknown_tag_id.c b/tests/core/proto/test_proto_get_light_state_unknown_tag_id.c
--- a/tests/core/proto/test_proto_get_light_state_unknown_tag_id.c
+++ b/tests/core/proto/test_proto_get_light_state_unknown_tag_id.c
@@ -112,7 +112,7 @@
             "\"_vendor\":null,"
             "\"hsbk\":[0,0,1,4000],"
             "\"power\":false,"
-            "\"label\":\"05:04:03:02:01:00\","
+            "\"label\":\"050403020100\","
             "\"tags\":[\"vapor\",\"d^-^b\"]"
         "},"
         "{"
